### PR TITLE
[SYCL-MLIR] Merge from intel/llvm sycl branch

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Please only review https://github.com/intel/llvm/pull/7667/commits/0d369c39c17a1e24b4c112c75ed96ed3e7d632a2.

1. https://github.com/intel/llvm/commit/aef89c8b41847c7235e00fbf8b2a2f10cb1aad25 [mlir] Cleanup lingering problems surrounding attribute/type aliases
    - problem: lit failures
    - solution: update CHECK lines to use the alias, add `-mlir-print-local-scope` for `types.mlir`.
2. `operandsMutable` renamed to `getOperandsMutable`
3. https://github.com/intel/llvm/commit/64f5fedb59e82a79c0c669c3d6591ca9eadb82fb [ASTContext][NFC] Remove getTargetAddressSpace(Qualifiers Q)
    - problem: build failures
    - solution: use `getAddressSpace()`

Please do not to squash and merge this PR.

### CODE REVIEW

**Comment by etiotto** [comment]
Please do not merge this massive PR before we have a chance to merge PR #7663

**Review by victor-eds — APPROVED** [review]

**Comment by etiotto** [comment]
@whitneywhtsang Merge this massive PR, then you can help with merging `https://github.com/intel/llvm/pull/7663` :-)

**Comment by whitneywhtsang** [comment]
> 

Resolved the merge conflict in https://github.com/intel/llvm/pull/7663.

**Pull request merged**
